From 2f4382f0d22639c433029dc2a797a60cdaeddd2e Mon Sep 17 00:00:00 2001
From: Alexei Kornienko <alexei.kornienko@gmail.com>
Date: Fri, 4 Oct 2013 10:51:19 +0300
Subject: [PATCH] Added trace header forwarding

---
 novaclient/client.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/novaclient/client.py b/novaclient/client.py
index ad594dd..3125f4d 100644
--- a/novaclient/client.py
+++ b/novaclient/client.py
@@ -10,6 +10,7 @@ OpenStack Client interface. Handles the REST calls and responses.
 import logging
 import os
 import time
+import tomograph
 
 import requests
 
@@ -206,6 +207,7 @@ class HTTPClient(object):
         # re-authenticate and try again. If it still fails, bail.
         try:
             kwargs.setdefault('headers', {})['X-Auth-Token'] = self.auth_token
+            tomograph.add_trace_info_header(kwargs['headers'])
             if self.projectid:
                 kwargs['headers']['X-Auth-Project-Id'] = self.projectid
 
-- 
1.8.1.2

