From 54495c5db007ceb31d6c1af1fd461e4025015d67 Mon Sep 17 00:00:00 2001
From: Alexei Kornienko <alexei.kornienko@gmail.com>
Date: Tue, 8 Oct 2013 10:57:33 +0300
Subject: [PATCH] Added trace header forwarding

---
 keystoneclient/httpclient.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/keystoneclient/httpclient.py b/keystoneclient/httpclient.py
index dc500db..2dc3b82 100644
--- a/keystoneclient/httpclient.py
+++ b/keystoneclient/httpclient.py
@@ -23,6 +23,7 @@ OpenStack Client interface. Handles the REST calls and responses.
 
 import copy
 import logging
+import tomograph
 import urlparse
 
 import requests
@@ -643,6 +644,7 @@ class HTTPClient(object):
             url_to_use = self.management_url
 
         kwargs.setdefault('headers', {})
+        tomograph.add_trace_info_header(kwargs['headers'])
         if self.auth_token:
             kwargs['headers']['X-Auth-Token'] = self.auth_token
 
-- 
1.8.1.2

